internal/fs/inode/dir.go

```diff
@@ -689,11 +690,15 @@ func (d *dirInode) readObjects(
 		}
 	}()
 
+	unsupportedObjects := []string{}
 	for _, o := range listing.MinObjects {
 		// Skip empty results or the directory object backing this inode.
 		if o.Name == d.Name().GcsObjectName() || o.Name == "" {
 			continue
 		}
+		if storageutil.IsUnsupportedObjectName(o.Name) {
```

Comment:
Do we need this here, I understand  - this will happen for implicit-dir scenario?

Reply:
it's possible, but it's a rare scenario that I am not able to recreate. So I am removing the check from listing.MinObjects .